### PR TITLE
Add WithTraceInfo that returns the logger with trace ID from the request

### PR DESCRIPTION
**Pull request by mariash**
Add a helper method to lager that returns a logger with trace information parsed from the request. We are planning to use this method across the components so it would be nice for logger to have it.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -12,6 +12,7 @@ require (
 	github.com/go-task/slim-sprig v0.0.0-20210107165309-348f09dbbbc0 // indirect
 	github.com/google/go-cmp v0.5.9 // indirect
 	github.com/google/pprof v0.0.0-20210407192527-94a9f03dee38 // indirect
+	github.com/openzipkin/zipkin-go v0.4.1 // indirect
```
[the remark this answers is not in the document]
Comment (mariash) [reply go.mod:15]: no, thanks!

**Comment by ctlong** [comment]
@mariash some thoughts:

**❓Are you sure that this is the place to put this logic?**
This logic seems very specific to an individual use-case for lager with zipkin in Cloud Foundry. Maybe it would be better to include it in a separate dependency that uses lager to emit the log line.

**❓Are you intending to support more headers?**
Word on the grapevine is that there may be more headers to support down the line, and if that's the case it would be nice if this method was setup to handle that right now.

**❓Is everything that's going to be doing tracing happening in the context of an HTTP request?**
It seems feasible that somewhere you may want to set `trace-id` or `span-id` properties that aren't contained in an `http.Request`. If so, it would be nice to make the input more generic and provide helper functions that do some of the trace ID / span ID extraction from `http.Request`.

**Comment by mariash** [comment]
@ctlong 

❓Are you sure that this is the place to put this logic?

We thought it would be nice to have it here, since we will use this method in every request handler across different components. We can make a separate package that extends lager logger if there is a preference to not mix this into this package. I see the point that it is more specific to CF than other methods.

❓Are you intending to support more headers?

No we don't intend to support more headers.

❓Is everything that's going to be doing tracing happening in the context of an HTTP request?

So far I have not seen a use case for the request context. We usually have an http handler, that sets up logger session and then this logger is being passed in to other methods. If we happen to need it in the future we can modify the method to look up in the context and then in the header.

**Comment by mariash** [comment]
I am closing this PR, I think it makes sense to have this in a separate package since this is very CF specific method

**Pull request merged**